### ISSUE TITLE
AI Rulers severely underestimate elves

### ISSUE THREAD

**Issue by JukeJukestar** [MattTheLegoman/RealmsInExile#625]
![image](https://user-images.githubusercontent.com/76008882/215453931-d0e8a788-b560-4dcd-8993-6753c6fbc401.png)

here we can see 3.8k levy and skirmishers thinking they can take on galadriel a few years into the game.
I think the problem here is that the AI severely underestimates army quality or doesnt consider it in the first place.
Whenever i am siegeing as an elf, the AI keeps smashing their armies into mine and being destroyed horribly because they think that they will win just because they have a higher number

**Comment by jj248** [comment]
Yes... this was always something we were worried about and wanted tester input on before finalising the levy size reduction malus / strength bonus.  Unfortunately, AI does not and cannot take full account of MaA quality and knight effectiveness when declaring war (PoD have a similar issue and asked about it at PDXCon - not much urgency to change from PDX).  

I'm _more_ surprised about the army behaviour, as that *should* be smart enough to take combat success likelihood into account.  I'd be interested if you were to tag-switch in the above case and go to attack the elves, what would the battle prediction be, as the AI should be aware of those factors.  I wonder if it *knows* this, but doesn't see any other options available.  

Other thing that might be relevant is that I know @Vierwood changed AI behaviour a little to slightly prefer battles more often than sieges, or something like that?  Can you comment Vier?

If this is consistently an issue, which I thought it might be, we only really have two options:

1. Further reduce or remove the levy size / unit strength modifiers to be closer to humans.  Pros: simple, AI knows how to handle this much more effectively.  Cons: loss of flavour for elves having small numbers but high strength.  We could do this anyway as a way to test army behaviour.

2. Try to influence war dec likelihood vs elves to have it make a smarter assessment of elven strength.  Pros: keeps current flavour.  Cons: I've found AI weights for CBs super fickle in the past and we might only be able to get it to behave sensibly by doing something to actually give them a 0% chance (e.g. don't even consider attacking unless you outnumber them 2:1). We'd need to keep up maintenance of these with any future CBs added.

With apologies for pings, also pinging @VectorMaximus given he's had thoughts about this in the past...

**Comment by jj248** [comment]
I've added an AI value that tweaks the weights used when determining what CBs to use.  This should now do the following where war target is an elf:

- Reduce weight by 75% when attacker is good-aligned
- Reduce weight by 25% when attacker is neutral-aligned
- Reduce weight to Zero when max military strength of attackers and allies is less than max military strength of defenders.
- Reduce weight by 75% when max military strength of attackers and allies is 1-2 times larger than max military strength of defenders.
- Reduce weight by 25% when max military strength of attackers and allies is 2-3 times larger than max military strength of defenders.
- Do not reduce weight when max military strength of attackers and allies is more than 3 times larger than max military strength of defenders.

Ready to test.  Testing should really be anecdotal here.  In your games, is the AI continuing to declare war on elves when they don't stand a chance?  Has the situation improved?

**Comment by Jauffr** [comment]
I have been running many observer games the past few days and have not seen any odd declarations of war on elves. Every war I have seen declared on elves arears to be what I consider at least even, if not vastly in favor of the human aggressors. I think this can be closed

**Comment by Jauffr** [comment]
I'm closing this. I do many observer games and have seen nothing like this happen anymore
